src/features/courses/course-controller.ts

```diff
@@ -2522,6 +2634,243 @@ class CourseController {
             throw new WrappedError('Could not create new student grade', e);
         }
     }
+
+    // does this actually belong in course-repository?
```

Comment:
typically you'll have a controller method and a repository method, the content of this would go in the repository and a method just calling the repository would be here

The reason for this is to keep a clean responsibility chain
Routes ==> Controller ==> Repository

Also it gives a centralized point if we wanted to do application logic on the create (i.e. check for fields maybe)